GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1344,7 +1344,7 @@ private void OnGridViewCellMouseDown(object sender, DataGridViewCellMouseEventAr
             var pt = _gridView.PointToClient(Cursor.Position);
             var hti = _gridView.HitTest(pt.X, pt.Y);
 
-            if (_latestSelectedRowIndex == hti.RowIndex)
+            if (_latestSelectedRowIndex == hti.RowIndex && _gridView.Rows[_latestSelectedRowIndex].Selected)
```

Comment:
This is basically "handle rightclick as selecting the revision". 
That the Selected property deviates from _latestSelectedRowIndexis an error that should be handled. Maybe this is sufficient.

This should be fine, but I cannot see that _gridView.Rows[_latestSelectedRowIndex] may not give null exceptions, so that should be checked.

However, it is not a guarantee that _latestSelectedRowIndex is still correct. There may have to be a check in CompareToBranchToolStripMenuItem_Click() and/or in OnGridViewSelectionChanged() or ContextMenuOpening().

Reply:
`_gridView.Rows[_latestSelectedRowIndex]` is de-referenced on line 1357 with a guard `IsValidRevisionIndex(_latestSelectedRowIndex)` but I wouldn't imagine that  `_latestSelectedRowIndex == hti.RowIndex` beeing true and not `_latestSelectedRowIndex >= 0 && _latestSelectedRowIndex < _gridView.RowCount` since `_latestSelectedRowIndex ` is equal to ` hti.RowIndex` which basicaly is an existing row (it has been clicked). Do you think I really should check for null or `IsValidRevisionIndex` ?

Reply:
With this fix I was not able to open the conext menu without having a row selected. I think that context menu scenarios are fixed.

But action exposed by the context menu can also be triggered whit hot keys when there is no selection in the grid. If I hit `CTRL+D` (Which is Campare to working directory) without a section the app keep crashing ! => In this case what should be the expected behavior ?

Reply:
No exception with hot keys - a null check is needed.